src/relayFeeCalculator/chain-queries/svmQuery.ts

```diff
@@ -137,65 +130,72 @@ export class SvmQuery implements QueryInterface {
+      mint,
+      mintInfo.programAddress
+    );
+    const relayerAta = await getAssociatedTokenAddress(
+      SvmAddress.from(relayer.toBase58()),
+      mint,
+      mintInfo.programAddress
+    );
+    const fillStatus = await getFillStatusPda(this.spokePoolAddress.toV2Address(), deposit, deposit.destinationChainId);
+    const eventAuthority = await getEventAuthority();
 
-    const [createTokenAccountsIx, approveIx, fillIx] = await Promise.all([
-      createTokenAccountsInstruction(mint, simulatedSigner),
-      createApproveInstruction(
-        mint,
-        deposit.outputAmount,
-        this.simulatedRelayerAddress,
-        this.spokePoolAddress,
-        mintInfo.data.decimals
-      ),
-      fillRelayInstruction(this.spokePoolAddress, deposit, simulatedSigner, associatedToken),
-    ]);
+    const relayData: SvmSpokeClient.FillRelayInput["relayData"] = {
+      depositor: toSvmAddress(deposit.depositor),
+      recipient: toSvmAddress(deposit.recipient),
+      exclusiveRelayer: toSvmAddress(deposit.exclusiveRelayer),
+      inputToken: toSvmAddress(deposit.inputToken),
+      outputToken: mint.toV2Address(),
+      inputAmount: deposit.inputAmount.toBigInt(),
+      outputAmount: deposit.outputAmount.toBigInt(),
+      originChainId: deposit.originChainId,
+      depositId: new Uint8Array(intToU8Array32(deposit.depositId.toNumber())),
+      fillDeadline: deposit.fillDeadline,
+      exclusivityDeadline: deposit.exclusivityDeadline,
+      message: new Uint8Array(Buffer.from(deposit.message, "hex")),
+    };
 
-    // Get the most recent confirmed blockhash.
-    const recentBlockhash = await this.provider.getLatestBlockhash().send();
-    const fillRelayTx = pipe(
-      createTransactionMessage({ version: 0 }),
-      (tx) => setTransactionMessageFeePayer(relayer.toV2Address(), tx),
-      (tx) => setTransactionMessageLifetimeUsingBlockhash(recentBlockhash.value, tx),
-      (tx) =>
-        isDefined(recipientCreateTokenAccountInstructions)
-          ? appendTransactionMessageInstructions(recipientCreateTokenAccountInstructions, tx)
-          : tx,
-      (tx) => appendTransactionMessageInstructions([createTokenAccountsIx, approveIx, fillIx], tx)
-    );
-    return fillRelayTx;
+    const simulatedSigner = SolanaVoidSigner(relayer.toBase58());
+    const fillInput: SvmSpokeClient.FillRelayInput = {
+      signer: simulatedSigner,
+      state,
+      delegate,
+      mint: mint.toV2Address(),
+      relayerTokenAccount: relayerAta,
+      recipientTokenAccount: recipientAta,
+      fillStatus,
+      tokenProgram: mintInfo.programAddress,
+      associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ADDRESS,
+      systemProgram: SYSTEM_PROGRAM_ADDRESS,
+      eventAuthority: eventAuthority,
+      program: this.spokePoolAddress.toV2Address(),
+      relayHash: relayDataHash,
+      relayData,
+      repaymentChainId: BigInt(repaymentChainId),
+      repaymentAddress: toSvmAddress(deposit.recipient),
```

Comment:
why deposit recipient is used as relayer repayment address?

Reply:
Nice catch, I've added an option repaymentAddress param that defaults to the simulatedRelayer